### PR TITLE
Implement API endpoints that shows config proposals in Supervisor [ECR-3610]

### PR DESCRIPTION
**Pull request by SergeiMal**
## Overview

---

See: https://jira.bf.local/browse/ECR-3610

### Definition of Done

- [ ] There are no TODOs left in the merged code
- [ ] Change is covered by automated tests
- [ ] Benchmark results are attached (if applicable)
- [ ] The [coding guidelines] are followed
- [ ] Public API has proper documentation
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

[coding guidelines]: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions

### CODE REVIEW

File: services/supervisor/src/api.rs [hunk 1]
```diff
@@ -34,6 +38,18 @@ pub trait PrivateApi {
     /// Creates and broadcasts the `StartService` transaction, which is signed
     /// by the current node, and returns its hash.
     fn start_service(&self, service: StartService) -> Result<Hash, Self::Error>;
+    /// Creates and broadcasts the `ConfigVote` transaction, which is signed
+    /// by the current node, and returns its hash.
+    fn confirm_config(&self, vote: ConfigVote) -> Result<Hash, Self::Error>;
```
Comment (popzxc) [inline services/supervisor/src/api.rs]: What about `create_config_proposal` endpoint?

File: services/supervisor/src/api.rs [hunk 2]
```diff
@@ -59,6 +75,25 @@ impl PrivateApi for ApiImpl<'_> {
...
+
+    fn confirm_config(&self, vote: ConfigVote) -> Result<Hash, Self::Error> {
+        self.broadcast_transaction(vote).map_err(From::from)
+    }
+}
+
+impl PublicApi for ApiImpl<'_> {
+    type Error = api::Error;
+
+    fn consensus_config(&self) -> Result<ConsensusConfig, Self::Error> {
```
Comment (popzxc) [inline services/supervisor/src/api.rs]: What about service configs?

File: services/supervisor/src/api.rs [hunk 3]
```diff
@@ -59,6 +75,25 @@ impl PrivateApi for ApiImpl<'_> {
...
+        Ok(CoreSchema::new(self.0.snapshot()).consensus_config())
+    }
+
+    fn config_proposal(&self) -> Result<ConfigProposalWithHash, Self::Error> {
+        Schema::new(self.0.instance.name, self.0.snapshot())
+            .pending_proposal()
+            .get()
+            .ok_or_else(|| Self::Error::NotFound("Pending config not found.".to_string()))
```
Comment (popzxc) [inline services/supervisor/src/api.rs]: I don't think that lack of pending proposal should be treated like error. Maybe `Ok(None)`?

File: services/supervisor/src/api.rs (hunk 2 above)
Comment (SergeiMal) [reply services/supervisor/src/api.rs]: It seems that the concrete service have to care about it by itself. Exonum doesn't store this information now.

Comment (popzxc) [reply services/supervisor/src/api.rs]: OK then.

**Pull request merged**